<launch>
    <!-- ========================================= -->
    <arg name="use_sim_time" default="true"/>
    <arg name="log_level" default="warn"/>

    <!-- ========================================= -->
    <node pkg="modularized_particle_filter" exec="predictor" name="predictor" output="screen" args="--ros-args --log-level warn">
        <remap from="twist" to="/kalman/twist" />
        <remap from="initialpose" to="/initialpose" />
        <remap from="pose" to="/init/pose"/> 

        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="num_of_particles" value="500" />
        <param name="resampling_interval_seconds" value="1.5" />
        <param name="prediction_rate" value="50.0" />

        <param name="static_linear_covariance" value="0.09" />
        <param name="static_angular_covariance" value="0.01" />
    </node>

    <!-- ========================================= -->
    <node pkg="modularized_particle_filter" exec="particle_visualize" name="particle_array_to_marker_array" output="screen" args="--ros-args --log-level $(var log_level)">
        <remap from="/particle_array" to="/predicted_particles" />
        <!-- <remap from="/particle_array" to="/gnss/weighted_particles" /> -->
        <!-- <remap from="/particle_array" to="/camera/weighted_particles" /> -->
        <remap from="/marker_array" to="/camera/weighted_particles/marker_array" />
    </node>
</launch>